### DISCUSSION TITLE
Handling "ERROR:paramiko.transport:Socket exception" using the "OpenSky Impala shell"

### DISCUSSION THREAD

**Discussion by yweidmann** [xoolive/traffic#186] [Q&A]
Dear list

The traffic library works perfectly, and I am able to retrieve data from the OpenSky-Network using the "OpenSky Impala shell".

Unfortunately, I am getting at irregular intervals the following error:
- ERROR:paramiko.transport:Socket exception: An existing connection was forcibly closed by the remote host (10054)

Somehow, all the try-except-finally blocks I implemented are not catching this exception. The code only uses the two following calls:
- opensky.history( ... )
- opensky.flightlist( ... )

Question:
- Does anyone have an idea, how to capture this exception and restarting the current function again?
- Or any ideas, why this exception arises?

Looking forward to any hints, ideas and suggestions.

Thank you very much in advance,
Yvo

**Comment by xoolive** [comment]
There is the method `opensky._connect()` that is used to connect to the shell (or to reconnect). You may try to use it if your process has been on for a long time and the connection might be stalled.

**Reply by yweidmann** [reply]
Hi xoolive
Thank you very much for the hint. I am testing it and it looks good. I will let run the code for some days and coming back with the experience made.

**Reply by yweidmann** [reply]
Dear xoolive

With using the method `opensky._connect()` before each opensky-function (`opensky.history()` and `opensky.flightlist()`) the code is running since about five days without any errors.

Looks like the `opensky._connect()` solved the problem.

Great.

Many thanks,
Yvo
